irohad/network/consensus_gate.hpp

```diff
@@ -1,52 +1,71 @@
 /**
- * Copyright Soramitsu Co., Ltd. 2017 All Rights Reserved.
- * http://soramitsu.co.jp
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *        http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
  */
 
 #ifndef IROHA_CONSENSUS_GATE_HPP
 #define IROHA_CONSENSUS_GATE_HPP
 
 #include <rxcpp/rx.hpp>
 
+#include "consensus/yac/yac_hash_provider.hpp"
+#include "ordering/on_demand_os_transport.hpp"
+
 namespace shared_model {
   namespace interface {
     class Block;
-  }
+    class Proposal;
+  }  // namespace interface
 }  // namespace shared_model
 
 namespace iroha {
   namespace network {
+    /// Current pair is valid
+    struct PairValid {
+      std::shared_ptr<shared_model::interface::Block> block_;
+    };
+
+    /// Network votes for another pair and round
+    struct VoteOther {
+      consensus::yac::YacHash hash;
+    };
+
+    /// Reject on proposal
+    struct ProposalReject {};
+
+    /// Reject on block
+    struct BlockReject {
+      consensus::yac::YacHash hash;
```

Comment:
YacHash is an implementation of consensus, it should not be present in this file